hack/clean-deploy.sh

```diff
@@ -2,15 +2,19 @@
 
 # expect oc to be in PATH by default
 OC_TOOL="${OC_TOOL:-oc}"
-nodeSelector="$(${OC_TOOL} get performanceprofile ci -o=jsonpath='{.spec.nodeSelector}'  | awk -F'[/:]' '{print $2}')"
-mcp="$(${OC_TOOL} get mcp -l machineconfiguration.openshift.io/role=$nodeSelector -o name | awk -F "/" '{print $2}')"
 
-# Remove node label
-echo "[INFO]: Unlabeling worker nodes"
-nodes=$(${OC_TOOL} get nodes --selector="node-role.kubernetes.io/${nodeSelector}" -o name)
-for node in $nodes
+profiles=$(${OC_TOOL} get profiles -o name)
+for profileName in $profiles
 do
-    ${OC_TOOL} label $node node-role.kubernetes.io/${nodeSelector}-
+    nodeSelector="$(${OC_TOOL} get performanceprofile $profileName -o=jsonpath='{.spec.nodeSelector}'  | awk -F'[/:]' '{print $2}')"
+    mcps+=($(${OC_TOOL} get mcp -l machineconfiguration.openshift.io/role=$nodeSelector -o name | awk -F "/" '{print $2}'))
```

Comment:
can we `continue` here in case we have found the `worker` role? I'm a bit afraid that we might unlabel normal worker nodes and delete their MCP

Reply:
hm.. we are doing that inside loop `for profileName in $profiles`, so we should not touch any node that does not have label equal to `nodeSelector`..

Reply:
yes, but you can have a profile targeting the `worker` role, if want to tune just all worker nodes.

Reply:
ah, right.. but I just moved this part into the loop, so we had this issue previously.. :) 

We still need to delete extra labels somehow, how about adding 
`if [[ $nodeSelector -ne "worker" ]]`

Reply:
pls, see changes

Reply:
> we had this issue previously

previously we didn't delete all profiles but `ci` only ;) It's easier now to mess things up.

> if [[ $nodeSelector -ne "worker" ]]

works for me :+1: 

/lgtm
/approve

Reply:
> previously we didn't delete all profiles but `ci` only ;) It's easier now to mess things up.
 Yes, but we could delete worker label... :)